infra/src/main/kotlin/fi/fta/geoviite/infra/projektivelho/PVClient.kt

```diff
@@ -79,177 +79,132 @@ class ProjektiVelhoClient @Autowired constructor(
         logger.integrationCall("postXmlFileSearch",
             "fetchStartTime" to fetchStartTime, "startOid" to startOid)
         val json = searchJson(fetchStartTime, startOid, 100)
-        return velhoClient
-            .post()
-            .uri("$XML_FILE_SEARCH_PATH?tagi=aineisto")
-            .headers { header -> header.setBearerAuth(fetchAccessToken(Instant.now())) }
-            .body(BodyInserters.fromValue(json))
-            .retrieve()
-            .bodyToMono<PVApiSearchStatus>()
-            .block(defaultBlockTimeout) ?: throw IllegalStateException("Projektivelho search failed")
+        return postMandatoryReturn<String, PVApiSearchStatus>("$XML_FILE_SEARCH_PATH?tagi=aineisto", json)
     }
+
     fun fetchVelhoSearchStatus(id: PVId): PVApiSearchStatus {
         logger.integrationCall("fetchVelhoSearchStatus", "id" to id)
-        return velhoClient
-            .get()
-            .uri("$XML_FILE_SEARCH_STATE_PATH/$id")
-            .headers { header -> header.setBearerAuth(fetchAccessToken(Instant.now())) }
-            .retrieve()
-            .bodyToMono<PVApiSearchStatus>()
-            .block(defaultBlockTimeout)
-            ?: throw IllegalStateException("Fetching running search status from ProjektiVelho failed: id=$id")
+        return getMandatory<PVApiSearchStatus>("$XML_FILE_SEARCH_STATE_PATH/$id")
```

Comment:
Nämä kutsuketjut oli kaikki melkein identtiset joten tein funktiot niille ja muutin one-linereiksi nämä käyttöpaikat

Reply:
Joo, tämä on hyvä refakki. Itsellekin tuli mieleen noita kirjoittaessa että nuohan ovat about identtisiä keskenään, mutta en sitten koskaan lähtenyt yhdistelemään näitä jostain syystä